lib/rspec/support/object_formatter.rb

```diff
@@ -3,9 +3,23 @@ module Support
     # Provide additional output details beyond what `inspect` provides when
     # printing Time, DateTime, or BigDecimal
     module ObjectFormatter
+      class << self
+        attr_accessor :max_formatted_output_length
+      end
```

Comment:
Adding mutable state at the module level is always a bit of a code smell, and requires gymnastics such a the around hook in your test to isolate it.  Up to know the object formatter didn't have any state which is why it is just available as methods directly off of the module.  I think that instead of adding this module-level state, I'd advocate:

* Turning it into a class, that receives `max_formatted_output_length` as an initialization argument.
* The existing interface be preserved (as a facade of sorts) by keeping class methods that delegate to a default shared instance of the class.
* Then in tests it can just instantiate a different instance with a different `max_formatted_output_length` value, with no need to mutate it and reset it.